apps/src/templates/teacherDashboard/teacherSectionsRedux.js

```diff
@@ -610,6 +610,25 @@ export function isRosterDialogOpen(state) {
   return getRoot(state).isRosterDialogOpen;
 }
 
+export function oauthProvider(state) {
+  return getRoot(state).provider;
+}
+
+export function sectionCode(state, sectionId) {
+  return (getRoot(state).sections[sectionId] || {}).code;
+}
+
+export function sectionProvider(state, sectionId) {
+  if (isSectionProviderManaged(state, sectionId)) {
+    return oauthProvider(state);
+  }
+  return null;
+}
+
+export function isSectionProviderManaged(state, sectionId) {
+  return !!(getRoot(state).sections[sectionId] || {}).providerManaged;
+}
+
```

Comment:
I hadn't thought about this being a step towards Paul's vision. That actually makes a lot of sense to me. I may start taking this approach too in new code.